tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -338,18 +304,21 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
         /* send and receive with mismatched signature algs */
         verifying_conn->handshake_params.client_cert_sig_scheme = &test_scheme;
         test_scheme.hash_alg = S2N_HASH_SHA256;
-        test_scheme.sig_alg = S2N_SIGNATURE_ECDSA;
+        test_scheme.sig_alg = test_case->sig_scheme->sig_alg;
+        /* 0xFFFF is an invalid iana value */
         test_scheme.iana_value = 0xFFFF;
```

Comment:
Is this what we were trying to test? From my reading, it sounds like we just wanted to make sure that we wouldn't verify a signature that wasn't of the type agreed on in the handshake. So a valid signature scheme, just not the one we agreed on.

Reply:
I'll readily admit that I had some trouble following this test, but the previous behavior of this test did seem to be using the type that was agreed upon in the handshake?
https://github.com/aws/s2n-tls/blob/9fb55ec5944f1293680678b016a99824d16508bb/tests/unit/s2n_tls13_cert_verify_test.c#L267-L269

All of the handshake params are populated from the the sig_scheme inherited from the test case.

Reply:
Sure it's initially populating them the same, but this case was labeled "send and receive with mismatched signature algs" and was changing the sig_alg part of the scheme. The previous case was labeled "Send and receive with mismatched hash algs" and changed the hash_alg part of the scheme.

But then this case also sets an invalid iana, and you're now explicitly checking for S2N_ERR_INVALID_SIGNATURE_SCHEME. I don't think you've made the test more wrong, but I'm wondering if it was wrong to begin with.

What happens if you remove the "test_scheme.iana_value = 0xFFFF" line?

Reply:
> What happens if you remove the "test_scheme.iana_value = 0xFFFF" line?
Oh, AND set test_scheme.sig_alg to something else. It's supposed to NOT match test_case->sig_scheme->sig_alg, if I'm understanding the test right.

Reply:
> Oh, AND set test_scheme.sig_alg to something else. It's supposed to NOT match test_case->sig_scheme-sig_alg, if I'm understanding the test right.

I can definitely do that, but that was the opposite of how the test case was previously written. If you change the `sig_alg` then the 
```
        EXPECT_SUCCESS(s2n_tls13_cert_verify_send(verifying_conn));
```
fails.

Reply:
For the "wrong hash" case, the test swaps out the hash_alg after the send, before the recv. Maybe that's what this test should be doing?

I'm just not sure what you're trying to test with your current version.